include/links-domain.php

```diff
@@ -99,11 +99,20 @@ public function get_hosts() {
 		$hosts = array();
 		foreach ( $this->options['domains'] as $lang => $domain ) {
 			$host = wp_parse_url( $domain, PHP_URL_HOST );
+
+			if ( ! is_string( $host ) ) {
+				continue;
+			}
+
 			// The function idn_to_ascii() is much faster than the WordPress method.
 			if ( function_exists( 'idn_to_ascii' ) ) {
 				// The use of the constant is mandatory in PHP 7.2 and PHP 7.3 to avoid a deprecated notice.
 				$hosts[ $lang ] = defined( 'INTL_IDNA_VARIANT_UTS46' ) ? idn_to_ascii( $host, 0, INTL_IDNA_VARIANT_UTS46 ) : idn_to_ascii( $host );
+			} elseif ( class_exists( 'WpOrg\Requests\IdnaEncoder' ) ) {
+				// Since WP 6.2.
+				$hosts[ $lang ] = \WpOrg\Requests\IdnaEncoder::encode( $host );
 			} else {
+				// Backward compatibility with WP < 6.2.
 				$hosts[ $lang ] = Requests_IDNAEncoder::encode( $host );
 			}
```

Comment:
Should we even keep this? Initially, idn conversion is useful for very few users. Maybe we can require that intl extension is activated for ths who need this conversion...

Reply:
You mean this?
```php
if ( function_exists( 'idn_to_ascii' ) ) {
	$hosts[ $lang ] = defined( 'INTL_IDNA_VARIANT_UTS46' ) ? idn_to_ascii( $host, 0, INTL_IDNA_VARIANT_UTS46 ) : idn_to_ascii( $host );
} else {
	$hosts[ $lang ] = $host;
}
```
I'm not sure why `idn_to_ascii()` would be needed, and so, how to know when it is needed?